### ISSUE TITLE
Fakespot [iPad] - opt in card is displayed before tapping on price tag

### ISSUE THREAD

**Issue by dragosb01** [mozilla-mobile/firefox-ios#19346]
## Steps to reproduce
1. Navigate to a product detail page on amazon.com
2. Navigate to the 'Settings' section and tap the "turn off review quality check" button
3. In a new tab, navigate to a product detail page on amazon.com
### Expected behavior
The detail page of the product should load without any other views
### Actual behavior
The detail page of the product loads and in the same time the opt in card
### Device & build information
* Device: iPad Air(5th generation), iOS 17.2
* Build version: 126.0
* First seen version: Yes
### Notes
Test case: https://testrail.stage.mozaws.net/index.php?/cases/view/2358864
Automation test affected: XCUITests/FakespotTests#testTurnOffAndOnTheReviewQualityCheck
Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/134391433/4d431933-2bad-4954-add5-234e63c5ac95

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Joy Chen

**Comment by MattLichtenstein** [comment]
Is this available for contributors? If so, I would like to work on it 🙂

**Comment by thatswinnie** [comment]
@MattLichtenstein yes, feel free to work on it. Please message me on Element if you have any questions.

**Comment by MattLichtenstein** [comment]
Thanks @thatswinnie 

This issue seems to occur after pressing any button in any of the many card views (onboarding, quality determination, product ad) in `FakespotViewController` that normally dismisses the view controller. This issue does not occur when using the "X" button or tap-outside/swipe gestures to dismiss the modal.

Would it be correct to assume that the expected behavior here is to _**never**_ have the `FakespotViewController` be presented without prior user interaction?

**Comment by thatswinnie** [comment]
@MattLichtenstein on iPad there are many scenarios where Fakespot should be restored/opened up on a product page. It should not be restored however if the user dismisses the view controller via the close button, when turning off Fakespot or when tapping outside the modal.

**Comment by MattLichtenstein** [comment]
@thatswinnie thank you for the clarification, though I do still have a couple of outstanding questions:

1. What about if they choose "not now" from the opt-in view? Or any of the various links (i.e. privacy notice, terms of use, etc) Currently, those result in the view controller being presented the next time a product page is loaded.
2. In what cases _should_ the view controller automatically restore

Thanks again for the support!

**Comment by thatswinnie** [comment]
@MattLichtenstein 

1. tapping on "not now" during Fakespot onboarding should be handled like closing Fakespot. If you tap on any of the other links in onboarding the sidebar/modal should be restored on iPad if you visit a product page again.
2. The sidebar/modal should be restored on iPad after the user opened Fakespot and didn't dismiss/close it through the previously described ways.

**Comment by MattLichtenstein** [comment]
@thatswinnie ahh that all makes sense. I guess my final question is, should we be waiting for the product details page to fully load before showing the modal/sidebar (unlike in the video from the issue description)?

Apologies for all of the questions - this issue seems to have a little more depth than I originally thought.

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v9000 (41169), with iPad Air 3rd gen (17.3).

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v127 (41813), with iPad Air 3rd gen (17.5).
